cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/to/SearchResponseFactory.java

```diff
@@ -0,0 +1,83 @@
+package com.devonfw.cobigen.api.util.to;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.devonfw.cobigen.api.exception.CobiGenRuntimeException;
+import com.devonfw.cobigen.api.exception.RESTSearchResponseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import jakarta.ws.rs.ProcessingException;
+
+/**
+ * Handles the responses from various search REST API's
+ *
+ */
+public class SearchResponseFactory {
+
+  /** Logger instance. */
+  static final Logger LOG = LoggerFactory.getLogger(SearchResponseFactory.class);
+
+  /**
+   * Gets a list of available search REST APIs (register a new search interface type here)
+   *
+   * @return list of available {@link SearchResponse}
+   */
+  static List<SearchResponse> getAvailableSearchInterfaces() {
```

Comment:
Since this list is only needed for this factory, this should not be a getter and instead just be a non-static constant as part of the factory, unless I am overlooking something.

Reply:
I've converted it to a constant now.